.github/workflows/build-msvc.yml

```diff
@@ -74,11 +74,11 @@ jobs:
           key: cygwin-packages
 
       - name: Compute a key to cache configure results
-        shell: bash
+        shell: bash -eo pipefail -o igncr {0}
```

Comment:
I think this can instead be set as the default? (see [docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/setting-a-default-shell-and-working-directory)?

From a previous conversation with @shym, the `PRISTINE_PATH` thing isn't needed and could be removed - cf. https://github.com/dra27/ocaml/pull/173

Reply:
Nice. I also tried setting `SHELLOPTS`, but it [triggers problems in other scripts](https://github.com/cygwin/cygwin-install-action/pull/22). Maybe `BASHOPTS` would work instead. I'll use your solution now.